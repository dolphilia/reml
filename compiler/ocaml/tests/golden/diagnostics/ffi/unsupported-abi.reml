extern "C" {
  #[target("x86_64-pc-windows-msvc")]
  #[link_name("ffi_demo_c")]
  #[ownership("borrowed")]
  #[calling_convention("system_v")]
  fn ffi_demo() -> i32
}
