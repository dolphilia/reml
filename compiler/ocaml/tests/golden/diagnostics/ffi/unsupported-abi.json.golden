{
  "diagnostics": [
    {
      "audit": {
        "bridge": {
          "status": "error",
          "block_target": "x86_64-pc-windows-msvc",
          "link_name": "ffi_demo_c",
          "expected_abi": "msvc",
          "extern_name": "ffi_demo",
          "target": "x86_64-pc-windows-msvc",
          "arch": "x86_64",
          "abi": "system_v",
          "ownership": "borrowed",
          "extern_symbol": "ffi_demo_c",
          "return": {
            "ownership": "borrowed",
            "wrap": "wrap_foreign_ptr",
            "release_handler": "none",
            "rc_adjustment": "none"
          }
        },
        "bridge.source_span": { "start": 0, "end": 168 }
      },
      "extensions": {
        "bridge": {
          "block_target": "x86_64-pc-windows-msvc",
          "link_name": "ffi_demo_c",
          "expected_abi": "msvc",
          "extern_name": "ffi_demo",
          "target": "x86_64-pc-windows-msvc",
          "arch": "x86_64",
          "abi": "system_v",
          "ownership": "borrowed",
          "extern_symbol": "ffi_demo_c",
          "return": {
            "ownership": "borrowed",
            "wrap": "wrap_foreign_ptr",
            "release_handler": "none",
            "rc_adjustment": "none"
          }
        }
      },
      "notes": [
        "指定された値: `system_v`", "要求される ABI: msvc",
        "サポートされる値: system_v, msvc, darwin_aapcs64"
      ],
      "domain": "型システム",
      "code": "ffi.contract.unsupported_abi",
      "severity": "error",
      "message": "外部関数 `ffi_demo` の ABI 契約がターゲットと整合していません",
      "location": {
        "file": "<入力>",
        "line": 1,
        "column": 1,
        "endLine": 1,
        "endColumn": 169
      }
    }
  ]
}
